### PR TITLE
Security framework for the API

### PR DESCRIPTION
**Pull request by blabno**

### CODE REVIEW

File: api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class TokenRegistry {
+
+    public static final long TTL = 30 * 60 * 1000;
+
+    private final TimeProvider timeProvider;
+    private final RandomStringGenerator randomStringGenerator;
+    private Map<String, Long> tokens = new HashMap<>();
+
+    public TokenRegistry() {
+        this(() -> UUID.randomUUID().toString(), System::currentTimeMillis);
```
Comment (ManfredKarrer) [inline api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java:16]: I think you have to use SecureRandom for cryptographically secure randomness.

File: api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public TokenRegistry(RandomStringGenerator randomStringGenerator, TimeProvider timeProvider) {
+        this.timeProvider = timeProvider;
```
Comment (ManfredKarrer) [inline api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java:20]: I think you have to use SecureRandom for cryptographically secure randomness.

File: api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+
+
+import com.github.javafaker.Faker;
```
Comment (ManfredKarrer) [inline api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java:28]: I dont find that class in the source code at https://github.com/Javafaker. 
Do we really need that? It is one of those examples where I don't want to get new dependencies (unknown developer, small functionality)

File: api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+
+    private String dataDir;
+    private BisqEnvironment bisqEnvironmentMock;
+    private TokenRegistry tokenRegistryMock;
+    private ApiPasswordManager apiPasswordManager;
+
+    private static String getRandomPasswordDifferentThan(String otherPassword) {
+        String newPassword;
+        do {
+            newPassword = Faker.instance().internet().password();
```
Comment (ManfredKarrer) [inline api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java:43]: What is that supposed to do? internet() is a weird method name...
As said above I prefer to not add that dependency.

File: api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java (hunk 3 above)
Comment (ManfredKarrer) [reply api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java:28]: Ah sorry, just saw its a test class. So then I dont care so much about that dependency anymore, but still would like to know better what the class provides.

File: api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java (hunk 4 above)
Comment (ManfredKarrer) [reply api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java:43]: Ah sorry, just saw its a test class. So then I dont care so much about that dependency anymore, but still would like to know better what the class provides.

**Review by ManfredKarrer — CHANGES_REQUESTED** [review]

File: api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java (hunk 3 above)
Comment (blabno) [reply api/src/test/java/bisq/api/http/service/auth/ApiPasswordManagerTest.java:28]: Faker is random data generator. On the top level it has methods with categories.

File: api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java (hunk 1 above)
Comment (blabno) [reply api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java:16]: `UUID.randomUUID` uses `SecureRandom` by default.

File: api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java (hunk 2 above)
Comment (blabno) [reply api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java:20]: Time provider is meant to provide actual time. It is used as abstraction for unit tests.

**Comment by blabno** [comment]
I've substituted `Random` with `SecureRandom`. I'm using the one created with default constructor.
I've tried `SecureRandom.getInstanceStrong()` but it takes about a minute to generate single long. So it's absolutely unacceptable performance.

File: api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java (hunk 1 above)
Comment (ManfredKarrer) [reply api/src/main/java/bisq/api/http/service/auth/TokenRegistry.java:16]: Ah ok.

**Review by ManfredKarrer — APPROVED** [review]
utACK

**Comment by ManfredKarrer** [comment]
I will merge master after the PR. Please update your branch afterwards to be in sync.

**Pull request merged**